wikia/common/mw_database/dbconfig.py

```diff
@@ -0,0 +1,108 @@
+import collections
+import os
+import random
+import yaml
+
+
+ConnectionDetails = collections.namedtuple('ConnectionDetails', ['hostname', 'username', 'password', 'dbname'])
+
+
+class DatabaseConfig(object):
+    def __init__(self, config_file, connect_fn, service_name=None):
+        if config_file is None:
+            config_file = os.environ['WIKIA_DB_YML']
+        with open(config_file) as fp:
+            ds_conf = yaml.load(fp)
+        self.mw_config = ds_conf[0]
+        self.mw_config["templateOverridesByCluster"] = self.mw_config.get("templateOverridesByCluster", {})
+        self.mw_config["templateOverridesByServer"] = self.mw_config.get("templateOverridesByServer", {})
+        self.mw_config["templateOverridesByService"] = self.mw_config.get("templateOverridesByService", {})
+        self.cluster_config = ds_conf[1]
+        self.external_cluster_config = ds_conf[2]
+        self.connect_fn = connect_fn
+        self.service_name = service_name
+
+    def get_connection_details(self, dbname, master=False, wc_master=False, override_db_name=None, username=None,
```

Comment:
Hmm... implementing `wfWaitForSlaves` is non-trivial given our current environment and qould require finding out whether the host is consul and if that's true then you would need to fetch list of actual hosts and replace the single entry with the list. Sounds like a good piece of work that I would like to avoid at this point.

I'd happily agree to the change what connection to `wikicities` master is done only when explicitly asked using `wc_master` instead of current implementation that also uses `master` for this purpose. @macbre Would it be good enough for you?